### PR TITLE
Move `torch.isinf` to C++

### PR DESCRIPTION
**Pull request by driazati**
As part of #30786, this moves `isinf` to C++. It's mostly the same as #28918.

### CODE REVIEW

File: aten/src/ATen/native/TensorCompare.cpp [hunk 1]
```diff
@@ -96,6 +96,10 @@ Tensor isfinite(const Tensor& self) {
   });
 }
 
+Tensor isinf(const Tensor& self) {
+  return at::native::isfinite(self).logical_not();
```
Comment (SsnL) [inline aten/src/ATen/native/TensorCompare.cpp]: Can we please use a kernel, similar to `isfinite`?

**Comment by eellison** [comment]
This looks good but I think we should probably try to land the PR from OSS contributor: https://github.com/pytorch/pytorch/pull/31099

File: aten/src/ATen/native/TensorCompare.cpp (hunk 1 above)
Comment (driazati) [reply aten/src/ATen/native/TensorCompare.cpp]: what’s the reason for that versus implementing this in terms of `isfinite`? is it just to not have to do the `logical_not`?

Comment (SsnL) [reply aten/src/ATen/native/TensorCompare.cpp]: 1 vs 2 kernels :)

**Pull request closed without merging**